src/components/FeatureViews/Accounts/UserProfile.js

```diff
@@ -236,146 +252,109 @@ const QUERIES = {
+    const AuthUser = useSelector(({ auth }) => auth);
 
-	const handleTabChange = (_, newTabIndex) => {
-		setTabIndex(newTabIndex);
-	};
+    // check if user is viewing a differnt profile
+    const classes = useStyles();
+    //
 
-	const tabs = () => {
+    // reset to first tab when profile changes
+    useEffect(() => {
+        setTabIndex(0);
+    }, [accountType, accountID]);
 
-		return (
-			<>
-				<Tabs
-					indicatorColor="primary"
-					onChange={handleTabChange}
-					textColor="primary"
-					value={tabIndex}
-				>
-					{displayTabs
-						.filter((tab) => (tab.access_permissions.includes(AuthUser.accountType)))
-						.map((tab) => (
-							tab.tab_id === 7 ?
-								<Tab
-									key={tab.tab_id}
-									label={
-										<>
-											{tab.icon}{tab.tab_heading}{importantNotesBadge}
-										</>
-									}
-								/>
-								:
-								<Tab
-									key={tab.tab_id}
-									label={
-										<>
-											{tab.tab_heading}
-										</>
-									}
-								/>
-						))}
-				</Tabs>
-				<ComponentViewer
-					inView={displayTabs.filter((tab) => (tab.access_permissions.includes(AuthUser.accountType)))[tabIndex]?.tab_id}
-					user={data}
-					id={accountID}
-					
-				/>
-			</>
-		);
-	};
+    // reset tab list when profile type changes
+    useEffect(() => {
+        setDisplayTabs(userTabs[accountType]);
+    }, [accountType]);
 
+    const { loading, error, data } = useQuery(QUERIES[accountType], {
+        variables: { ownerID: accountID },
+    });
 
+    if (loading) return <Loading />;
 
-	// useEffect(() => {
-	// 	if (data) {
-	// 		const numImportantNotes = Object.values(data.notes || {}).reduce(
-	// 			(total, { important }) => (important ? total + 1 : total),
-	// 			0
-	// 		);
-	// 		if (data.role !== "receptionist") {
-	// 			setDisplayTabs((prevTabs) => {
-	// 				const newTabs = [...prevTabs];
-	// 				const notesIndex = newTabs.findIndex((tab) => tab.tab_id === 7);
-	// 				newTabs[notesIndex] = {
-	// 					...newTabs[notesIndex],
-	// 					icon: (
-	// 						<Badge badgeContent={numImportantNotes} color="primary">
-	// 							<NoteIcon className="TabIcon" />
-	// 						</Badge>
-	// 					),
-	// 				};
-	// 				return newTabs;
-	// 			});
-	// 		}
-	// 	}
-	// }, [data]);
+    if (error) return <Redirect to="/PageNotFound" />;
 
+    const { notes } = data;
+    const numImportantNotes = notes.filter((note) => note.important).length;
+    const importantNotesBadge =
+        numImportantNotes > 0 ? numImportantNotes : null;
 
+    const handleTabChange = (_, newTabIndex) => {
+        setTabIndex(newTabIndex);
+    };
 
+    const tabs = () => {
+        return (
+            <>
+                <Tabs
+                    classes={{ indicator: classes.MuiIndicator }}
+                    onChange={handleTabChange}
+                    value={tabIndex}
+                >
+                    {displayTabs
+                        .filter((tab) =>
```

Comment:
since `(tab) => tab.access_permissions.includes(AuthUser.accountType)` is used in both this filter (line 311) and the next (line 337). set this function as a reuseable function. example: `const tabsInViewforAccountType = (accountType, tab) =>  tab.access_permissions.includes(AuthUser.accountType)`